### PR TITLE
fix: add search parameter to icons

### PR DESCRIPTION
**Pull request by Birkbjo**
Adds `search`-param to `/icons` API.

This allows searching for a part of the `iconkey`, or partial matches in any keyword.

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java [hunk 1]
```diff
@@ -210,6 +215,15 @@ private String buildIconQuery(
       parameterSource.addValue("keys", params.getKeys());
     }
 
+    if (params.hasSearch()) {
+      String searchValue = params.getSearch();
+
+      sql += hlp.whereAnd() + " c.iconkey ilike :search";
+      sql += " OR jsonb_path_exists(keywords, '$[*] ? (@ like_regex " + doubleQuote(searchValue) + ")')";
```
Comment (Birkbjo) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java:222]: I had ALOT of problems getting the correct escape params to work here. Whatever I tried I was not able to use the parameter source to put in the searchValue in the `jsonb` query.

I did intent to add start-of-string anchor to this, but I see I forgot about that. Any thoughts about that?

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java [hunk 2]
```diff
@@ -210,6 +215,15 @@ private String buildIconQuery(
...
+
+      parameterSource.addValue("search", wrap(addIlikeReplacingCharacters(searchValue), '%'));
```
Comment (Birkbjo) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java:224]: not sure if I should use `addIlikeReplacingCharacters`, but it seemed the safest choice. However this is in the `dataitem`-package, I'm not sure how strict you are on including code from other modules like that?

Comment (jbee) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java:226]: You might have to add the module dependency if it wasn't already there

**Review by jbee — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java (hunk 2 above)
Comment (Birkbjo) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java:226]: Yeah, vscode took care of that

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java (hunk 1 above)
Comment (Birkbjo) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java]: Alright, I managed to add the parameter using `parameterSource`. But I had to add doubleQuotes in the SQL itself for it to work.

**Pull request merged**